leveldb/src/main/java/org/iq80/leveldb/impl/Level.java

```diff
@@ -68,10 +68,14 @@ public int getLevelNumber()
     }
 
     @Override
-    public InternalIterator iterator()
+    public SeekingIterator<InternalKey, Slice> iterator()
     {
         if (levelNumber == 0) {
-            return new Level0Iterator(tableCache, files, internalKeyComparator);
+            ImmutableList.Builder<SeekingIterator<InternalKey, Slice>> builder = ImmutableList.builder();
+            for (FileMetaData file : files) {
+                builder.add(tableCache.newIterator(file));
+            }
+            return new MergingIterator(builder.build(), internalKeyComparator);
```

Comment:
MergingInterator#resetPriorityQueue is bogus. it should be fixed before using "MergingIterator" widely.

Reply:
Merged from master

Reply:
:thumbsup: